homework/index.js

```diff
@@ -28,26 +30,133 @@
     });
     return elem;
   }
+  function formatDate(dateString) {
+    const dateTime = new Date(dateString);
+    return dateTime.toLocaleString();
+  }
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+  function renderError(error) {
+    const container = document.getElementById('main-container');
+    container.innerHTML = '';
+    createAndAppend('div', container, {
+      text: error.message,
+      class: 'alert',
+    });
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, {
-          text: err.message,
-          class: 'alert-error',
+  function addTable(table, body, value) {
+    const tr = createAndAppend('tr', table);
+    createAndAppend('th', tr, { text: body });
+    createAndAppend('td', tr, { text: value });
+    return tr;
+  }
+
+  function renderRepoDetails(repo) {
+    const leftSide = document.getElementById('left-side');
+    leftSide.innerHTML = '';
+    const ul = createAndAppend('ul', leftSide);
+    const li = createAndAppend('li', ul);
+    const table = createAndAppend('table', li);
+    const tr1 = addTable(table, 'Repository:', '');
+    createAndAppend('a', tr1.lastChild, {
+      href: repo.html_url,
+      text: repo.name,
+    });
+    addTable(table, 'Description:', repo.description);
+    addTable(table, 'Fork: ', repo.forks);
+    addTable(table, 'Updated:', formatDate(repo.updated_at));
+  }
+
+  function fetchAndRender(repo) {
+    const rightSide = document.getElementById('right-side');
+    rightSide.innerHTML = '';
+    fetchJSON(repo.contributors_url)
+      .then(contributors => {
+        createAndAppend('span', rightSide, {
+          text: 'Contributions',
+          class: 'title',
         });
-        return;
-      }
-      const ul = createAndAppend('ul', root);
-      repos.forEach(repo => renderRepoDetails(repo, ul));
+        const contributorList = createAndAppend('ul', rightSide, {
+          class: 'contributor-list',
+        });
+        contributors.forEach(contributor => {
+          const list = createAndAppend('li', contributorList, {
+            class: 'contributor-item',
+          });
+          createAndAppend('img', list, {
+            class: 'image',
+            src: contributor.avatar_url,
+          });
+          createAndAppend('span', list, {
+            class: 'contributor-name',
+            text: contributor.login,
+          });
+          createAndAppend('span', list, {
+            class: 'contribution-count',
+            text: contributor.contributions,
+          });
+          list.addEventListener('click', () => {
+            window.open(contributor.html_url, '_blank');
+          });
+          list.addEventListener('key', event => {
+            if (event.key === 'Enter')
+              window.open(contributor.html_url, '_blank');
+          });
```

Comment:
This second event listener for the Enter key should allow the user to open the contributor's page. Unfortunately, there is no way for a user to navigate to the list item in order press Enter on it. You need to include the list item into the keyboard tab order. You can do this by setting the `tabindex` attribute when you create the list item in line 83:

```js
const list = createAndAppend('li', contributorList, {
  class: 'contributor-item',
  tabindex: 0,
});
```

But another approach to enable both keyboard and mouse click for opening the contributor's page without using any event listeners is to wrap the whole content of the list item into an `<a>` tag with an appropriate `href`. Responding to mouse clicks and the Enter key is standard behaviour for an `<a>` tag.

The proposed function `renderContributor()` would then look like this:

```js
function renderContributor(contributor, contributorList) {
  const li = createAndAppend('li', contributorList);
  const anchor = createAndAppend('a', li, {
    class: 'contributor-item',
    href: contributor.html_url,
    target: '_blank',
  });
  createAndAppend('img', anchor, {
    class: 'image',
    src: contributor.avatar_url,
  });
  createAndAppend('span', anchor, {
    class: 'contributor-name',
    text: contributor.login,
  });
  createAndAppend('span', anchor, {
    class: 'contribution-count',
    text: contributor.contributions,
  });
}
```

To get rid of the underline normally present for a `<a>` tag you could set the `text-decoration: none;` for the `contributor-item` class.